src/zenml/client.py

```diff
@@ -2786,6 +2786,43 @@ def delete_artifact(
         self.zen_store.delete_artifact(artifact_id=artifact.id)
         logger.info(f"Deleted artifact '{artifact.name}'.")
 
+    def prune_artifacts(
+        self, only_artifact: bool = False, only_metadata: bool = False
+    ) -> None:
+        """Delete all unused artifacts and artifact versions.
+
+        Args:
+            only_artifact: Only delete artifacts
+            only_metadata: Only delete artifact metadata
+
+        Raises:
+            RuntimeError: If any artifact cannot be deleted.
+        """
+        unused_artifact_versions = depaginate(
+            partial(self.list_artifact_versions, only_unused=True)
+        )
+
+        if not unused_artifact_versions:
+            logger.info("No unused artifact versions found.")
+            return
+
+        for unused_artifact_version in unused_artifact_versions:
+            try:
+                self.delete_artifact_version(
+                    name_id_or_prefix=unused_artifact_version.id,
+                    delete_metadata=not only_artifact,
+                    delete_from_artifact_store=not only_metadata,
+                )
+                unused_artifact = unused_artifact_version.artifact
+                if not unused_artifact.versions and not only_artifact:
+                    self.delete_artifact(unused_artifact.id)
```

Comment:
Similar to the PR where you implemented the "delete all model version artifacts", there is a performance hit with this type of loop that ends up calling the REST API multiple times. You should consider implementing this as a REST API endpoint and looping in the SQL zen store instead (perhaps even deleting entities en-masse through a single SQL query, if possible). This opens the possibility of using this functionality through the dashboard.

Reply:
Hey, I'm half way through implementing this change, but stuck at the `verify_permissions_and_delete_entity`. It was rather clear what to check on single entry deletion, but here I prune Artifact, Artifact Version and Data at scale. Any thoughts? Shall I check for Admin permissions to do so?

Reply:
I only have a half-baked idea about how to get around this: maybe we need a new `prune` permission that you check globally at server level, as opposed to checking against every resource. But that requires more work, not only here, but probably also at Cloud API level.

@schustmi what do you think ?